bin/ocaml/ocaml_merlin.ml

```diff
@@ -264,6 +291,14 @@ module Dump_dot_merlin = struct
             ~doc:
               "The path to the folder of which the configuration should be printed. \
                Defaults to the current directory.")
+    and+ ctx_name =
```

Comment:
I missed this at first, but I think this is where you should be parsing the context, instead of `arg_to_ctx`.

```ocaml
Arg.(value & opt Selected_context.conv Selected_context.Default & ...)
```

where `Selected_context.conv` is 


```ocaml
let conv =
  let parse ctx_name =
    match Context_name.of_string_opt ctx_name with
    | None -> User_error.raise [ Pp.textf "Invalid context name %S" ctx_name ]
    | Some ctx_name -> Custom ctx_name
  in
  let print = function Default -> "default" | Custom ctx -> ctx in
  Arg.conv ~docv:"context" (parse, print)
```

Reply:
Added in https://github.com/ocaml/dune/pull/10324/commits/666e5250015e19f5f5d391b7e35ac64e13e6963b.